internal/responses/responses.go

```diff
@@ -0,0 +1,27 @@
+package responses
+
+import "encoding/json"
+
+type Response struct {
+	Data    interface{} `json:"data"`
+	Message string      `json:"message"`
+	Status  int         `json:"status"`
+	Meta    interface{} `json:"_meta"`
+}
+
+func (r *Response) PrepareResponse() ([]byte, error) {
+	resp, err := json.Marshal(r)
+	if err != nil {
+		return nil, err
+	}
+	return resp, err
+}
+
+func NewResponse(data interface{}, msg string, status int, meta interface{}) *Response {
+	r := new(Response)
+	r.Data = data
+	r.Message = msg
+	r.Status = status
+	r.Meta = meta
+	return r
+}
```

Comment:
we don't really add anything to the table

```
r := new(response.Response)

r.Data = data
r.Message = msg
r.Status = status
r.Meta = meta

rrr, err := json.Marshal(r)
if err != nil {
               w.WriteHeader(http.ServerInternalError)
	      return 
}

w.Write(rr)
w.WriteHeader(http.StatusOK)

```

logic above is simple enough to be in the handler imho, hiding it behind the PrepareResponse simply adds more `ifs` and hids it , arguable point i know, i would stick with explicit for now